ProcessLib/Parameter/MeshElementParameter.h

```diff
@@ -0,0 +1,59 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef PROCESSLIB_MESHELEMENTPARAMETER_H
+#define PROCESSLIB_MESHELEMENTPARAMETER_H
+
+#include "Parameter.h"
+
+namespace MeshLib
+{
+template <typename T>
+class PropertyVector;
+}  // MeshLib
+
+namespace ProcessLib
+{
+/// A parameter represented by a mesh property vector.
+template <typename T>
+struct MeshElementParameter final : public Parameter<T> {
+    MeshElementParameter(MeshLib::PropertyVector<T> const& property)
+        : _property(property)
+        , _cache(_property.getNumberOfComponents())
+    {
+    }
+
+    unsigned getNumberOfComponents() const override
+    {
+        return _property.getNumberOfComponents();
+    }
+
+    std::vector<T> const& getTuple(double const /*t*/,
+                                   SpatialPosition const& pos) const override
+    {
+        auto const e = pos.getElementID();
+        assert(e);
+        auto const num_comp = _property.getNumberOfComponents();
+        for (std::size_t c=0; c<num_comp; ++c) {
+            _cache[c] = _property.getComponent(*e, c);
+        }
```

Comment:
Suggestion: Line 42 - 45 could be implemented as `PropertyVector::getAllComponents(...)`? If this would be helpful I could implement this.